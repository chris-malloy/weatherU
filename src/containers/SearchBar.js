import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { fetchWeather } from '../actions';
import { Row, Input, Button, Icon } from 'react-materialize';

export class SearchBar extends Component {
  constructor() {
    super();

    this.state = { term: '', error: '' };

    this.onInputChange = this.onInputChange.bind(this);
    this.onFormSubmit = this.onFormSubmit.bind(this);
  };

  onInputChange(e) {
    this.setState({ term: e.target.value })
  };

  onFormSubmit(e) {
    e.preventDefault();
    this.props.fetchWeather(this.state.term);
    this.setState({ term: '' });
  };

  render() {
    return (
      <Row>
        <form onSubmit={this.onFormSubmit}>
          <Input error="" s={12} label="search" onChange={this.onInputChange} value={this.state.term}></Input>
          <Button type="submit" style={{width: '100%'}} waves='light'><Icon center>search</Icon></Button>
          </form>
      </Row>
    )
  };
};

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ fetchWeather }, dispatch)
};

export default connect(null, mapDispatchToProps)(SearchBar);
