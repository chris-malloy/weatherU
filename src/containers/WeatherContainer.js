import React, { Component } from 'react';
import { connect } from 'react-redux';
import WeatherCards from '../components/WeatherCards';

export class WeatherContainer extends Component {
  render() {
    return (
      <div>
        <WeatherCards weatherList={[]} />
      </div>
    );
  };
};

export default connect()(WeatherContainer);
