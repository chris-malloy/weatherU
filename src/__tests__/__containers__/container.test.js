import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { SearchBar } from '../../containers/SearchBar';
import { WeatherContainer } from '../../containers/WeatherContainer';

configure({ adapter: new Adapter() });

describe('Containers', () => {
  describe('Search', () => {
    const searchBar = (<SearchBar />);

    it('renders correctly', () => {
      expect(searchBar).toMatchSnapshot();
    })
  });

  describe('WeatherContainer', () => {
    const weatherContainer = shallow(<WeatherContainer />);

    it('renders correctly', () => {
      expect(weatherContainer).toMatchSnapshot();
    });

    it('has a WeatherCards component', () => {
      expect(weatherContainer.find('WeatherCards').exists()).toBe(true);
    })
  });
});
