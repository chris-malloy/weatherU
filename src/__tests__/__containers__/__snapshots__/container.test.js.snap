// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers Search renders correctly 1`] = `<SearchBar />`;

exports[`Containers WeatherContainer renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WeatherContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WeatherCards
        weatherList={Array []}
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "weatherList": Array [],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WeatherCards
          weatherList={Array []}
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "weatherList": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
